main.py

```
@@ -450,6 +487,34 @@ async def cli_loop(sk, pk, chain, mempool, network):
+
+        # ── import_key ──
+        elif cmd == "import_key":
+            if len(parts) < 2:
+                print("  Usage: import_key <hex_private_key>")
+                continue
+            
+            try:
+                new_sk = SigningKey(bytes.fromhex(parts[1]))
+                new_pk = new_sk.verify_key.encode(encoder=HexEncoder).decode()
+                
+                path = datadir or "."
```

Ruff (0.15.20)

[error] Undefined name `datadir`

(F821)